src/controllers/expenses.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+const expenseService = require('../services/expenses');
+const userServices = require('../services/users');
+
+const getAll = (req, res) => {
+  const filterParams = req.query;
+
+  const expenses = expenseService.getAll();
+  const filteredExpenses
+    = expenseService.filterExpenses(expenses, filterParams);
+
+  res.send(filteredExpenses);
+};
+
+const getOne = (req, res) => {
+  const { expenseId } = req.params;
+
+  const expense = expenseService.getOne(+expenseId);
+
+  if (isNaN(+expenseId)) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(expense);
+};
+
+const addExpense = (req, res) => {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = req.body;
+
+  const user = userServices.getOne(+userId);
+
+  if ((typeof spentAt !== 'string')
+    || (typeof title !== 'string')
+    || (typeof amount !== 'number')
+    || (typeof category !== 'string')
+    || (note && typeof note !== 'string')
+    || !user) {
```

Comment:
if you have several conditions, will be better to move the result of them to separate boolean variable